### PR TITLE
adds section commenting status and inheritance

### PR DESCRIPTION
**Pull request by sheralim012**
- adds comment related fields in the section
- by default page comment status will be inherited
- if article commenting status is inherited then ancestors will be used to determine the commenting status
- updated content migration to handle section commenting fields

### CODE REVIEW

**Review by ehmadzubair — APPROVED** [review]

File: comments/models.py [hunk 1]
```diff
@@ -26,30 +29,57 @@ class CommentableMixin(models.Model):
...
+                return ancestor
+
     def should_show_comments_list(self):
-        return self.commenting_status in [CommentStatus.OPEN, CommentStatus.CLOSED, CommentStatus.TIMESTAMPED]
+        from home.models import Section
+
+        commenting_status = self.commenting_status
+        if commenting_status == CommentStatus.INHERITED:
+            ancestor = self._get_valid_ancestor_for_commenting()
+            commenting_status = ancestor.commenting_status
```
Comment (geoo89) [inline comments/models.py:58]: Check for None, e.g. if the whole ancesteor tree has INHERITED, the page crashes.
In that case, use 'closed' as a default. (Or check what V1 uses, might be not so easy to find out though.)

File: comments/models.py [hunk 2]
```diff
@@ -26,30 +29,57 @@ class CommentableMixin(models.Model):
...
-        return self.commenting_status == CommentStatus.OPEN or \
-               (self.commenting_status == CommentStatus.TIMESTAMPED and commenting_still_valid)
+        commenting_still_valid = True
+        commenting_status = self.commenting_status
+        commenting_starts_at = self.commenting_starts_at
+        commenting_ends_at = self.commenting_ends_at
+
+        if commenting_status == CommentStatus.INHERITED:
+            ancestor = self._get_valid_ancestor_for_commenting()
+            commenting_status = ancestor.commenting_status
```
Comment (geoo89) [inline comments/models.py:72]: same here.

**Review by geoo89 — COMMENTED** [review]
Looks good, except if the ancestor recursion bottoms out the page crashes.

**Comment by sheralim012** [comment]
@geoo89 fixed

File: comments/models.py (hunk 1 above)
Comment (sheralim012) [reply comments/models.py]: v1 was using disabled for it

**Pull request merged**